### DISCUSSION TITLE
Disable rule suppression via code?

### DISCUSSION THREAD

**Discussion by kenherring-rkt** [Riverside-Software/sonar-openedge#1018] [Q&A]
Is it possible to disable rule suppression in the code? Here's an example of what we don't want users doing, as we'd rather manage the issues directly in sonar via issue resolution.

```
{&_proparse_ prolint-nowarn(recidkeyword)}
define variable var as recid no-undo.
```

**Comment by gquerret** [comment]
It's currently not possible to disable the `{&_proparse_}` preprocessor. As a workaround, you can monitor usage of such preprocessor expressions in your codebase, as there's a measure for that (size -> proparse directives). You can even add a check in your quality gate for this measure.

**Comment by ccecvb** [comment]
We actually want almost the opposite, we normally don't want to manage this in Sonar, except for bulk wont-fix on certain legacy code
I don't mind a config switch to ignore the prolint-nowarn directives, but we need in source techniques to ignore the rules

**Reply by kenherring-rkt** [reply]
> We actually want almost the opposite, we normally don't want to manage this in Sonar, except for bulk wont-fix on certain legacy code I don't mind a config switch to ignore the prolint-nowarn directives, but we need in source techniques to ignore the rules

Yeah I wasn't suggesting getting rid of it, just wondering if there was an easy option.  I'm betting it would be pretty easy for us to write a custom rule for it though.
